compiler/src/iree/compiler/Codegen/Common/GPUPipelining.cpp

```diff
@@ -23,6 +25,49 @@ namespace iree_compiler {
 static const StringLiteral kPipeliningLoopMarker = "__pipelining_K_loop__";
 static const StringLiteral kPipeliningGlobalLoad = "__pipelining_global_load__";
 
+
+// Returns a new AsyncCopyOp with Zfill 
+static Operation* replaceAsyncCopywithAsyncCopyZfill(Operation* op, Value pred, PatternRewriter& rewriter) {
+
+  if (!isa<nvgpu::DeviceAsyncCopyOp>(op)) {
+    // Return the op if it is side effect free or barrier, commit group, or shared memory load
+    if (mlir::isSideEffectFree(op) || 
+        (isa<gpu::BarrierOp, nvgpu::DeviceAsyncCreateGroupOp, nvgpu::LdMatrixOp>(op)))
+      return op;
+    else 
+      assert(true && "Unpeeled epilogue not supported with a side-effect instruction with no predication.");
+  }
+
+  // replace mainloop AsyncCopy with AsyncCopy(zfill) inline asm.
+  auto asyncCopyOp = dyn_cast<nvgpu::DeviceAsyncCopyOp>(op);
+  auto loc = asyncCopyOp->getLoc();
+
+  // create srcElement Value based on the pred
+  // srcElement = (pred) ?  dstElements : 0;
+ // asyncCopyOp.getDstElements().getZExtValue()
+  Value dstElements = 
+      rewriter.create<arith::ConstantOp>(loc, asyncCopyOp.getDstElementsAttr());
+
+  Value c0Index =
+      rewriter.create<arith::ConstantIndexOp>(loc, 0);
+
```

Comment:
nit: consider removing the extra empty lines

Reply:
Done.